src/test/rgw/test_rgw_lua.cc

```diff
@@ -1613,6 +1613,35 @@ TEST(TestRGWLua, MemoryLimit)
   ASSERT_NE(rc, 0);
 }
 
+TEST(TestRGWLua, LuaRuntimeLimit)
+{
+  std::string script = "print(\"hello world\")";
+  
+  DEFINE_REQ_STATE;
+
+  // runtime should be sufficient
+  s.cct->_conf->rgw_lua_max_runtime_per_state = 1000; // 1 second runtime limit
+  int rc = lua::request::execute(nullptr, nullptr, nullptr, &s, nullptr, script);
+  ASSERT_EQ(rc, 0);
+
+  // no runtime limit
+  s.cct->_conf->rgw_lua_max_runtime_per_state = 0;
+  rc = lua::request::execute(nullptr, nullptr, nullptr, &s, nullptr, script);
+  ASSERT_EQ(rc, 0);
+  
+  // script should exceed the runtime limit
+  script = R"(
```

Comment:
please also try a the folowing script:
```lua
for i = 1, 10 do
  os.execute("sleep 1")
end
```
with 5sec limit

Reply:
calling `sleep()` won't happen in reasl life, but blocking IO may happen.
we probably need the vm instruction count to be a config option as well.
we can set to 1000 by defauls, but would need a way to change it depending with the usecase